modu/source_repository/source_repository.py

```diff
@@ -87,16 +88,23 @@ def _parse_ast(self, of_file: FileInfo) -> tree_sitter.Tree:
 
     def _discover_files(self) -> Dict[Path, FileInfo]:
         if self.src_root.is_file():
+            for ignore in self.ignored_glob_regex_list:
+                if self.src_root.match(ignore):
+                    return {}
             return {self.src_root: self._inspect_file(self.src_root)}
 
-        file_infos: Dict[Path, FileInfo] = self._discover_in_dir(self.src_root)
+        files = self._get_filtered_files(self.src_root)
+
+        file_infos: Dict[Path, FileInfo] = self._discover_in_dir(files)
+
         return file_infos
 
-    def _discover_in_dir(self, root_dir: Path) -> Dict[Path, FileInfo]:
+    def _discover_in_dir(self, files: list) -> Dict[Path, FileInfo]:
```

Comment:
Maybe _discover_in_dir isn't the best name anymore since it really just is a function that filters a list of files based on some language / encoding requirements.

I suggest `_inspect_files`